client/components/BotRunTestStatusList/index.js

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+    COLLECTION_JOB_STATUS_BY_TEST_PLAN_RUN_ID_QUERY,
+    TEST_PLAN_RUNS_TEST_RESULTS_QUERY
+} from './queries';
+import { useLazyQuery, useQuery } from '@apollo/client';
+import styled from '@emotion/styled';
+import ReportStatusDot from '../common/ReportStatusDot';
+import { isBot } from '../../utils/automation';
+
+const BotRunTestStatusUnorderedList = styled.ul`
+    list-style-type: none;
+    background-color: #f6f8fa;
+    font-size: 0.9rem !important;
+    padding: 0.5rem 0;
+    margin: 0.5rem 0;
+    white-space: nowrap;
+`;
+
+const BotRunTestStatusList = ({ testPlanReportId, runnableTestsLength }) => {
+    const {
+        data: testPlanRunsQueryResult,
+        startPolling,
+        stopPolling
+    } = useQuery(TEST_PLAN_RUNS_TEST_RESULTS_QUERY, {
+        variables: { testPlanReportId },
+        fetchPolicy: 'cache-and-network',
+        pollInterval: 2000
+    });
+
+    const [getCollectionJobStatus, { data: collectionJobStatusQueryResult }] =
+        useLazyQuery(COLLECTION_JOB_STATUS_BY_TEST_PLAN_RUN_ID_QUERY, {
+            fetchPolicy: 'cache-and-network'
+        });
+
+    const [collectedData, setCollectedData] = useState([]);
+    const requestedTestRunIds = useRef(new Set());
+
+    const filteredTestPlanRuns = useMemo(() => {
+        if (!testPlanRunsQueryResult?.testPlanRuns) {
+            return [];
+        }
+        return testPlanRunsQueryResult.testPlanRuns.filter(testPlanRun =>
+            isBot(testPlanRun.tester)
+        );
+    }, [testPlanRunsQueryResult?.testPlanRuns]);
+
+    useEffect(() => {
+        if (filteredTestPlanRuns && filteredTestPlanRuns.length > 0) {
```

Comment:
We can shrink the checks in this component by taking for granted that `filteredTestPlanRuns` always stores an array value.

Reply:
Nice!